src/Compilers/Core/Portable/Compilation/SymbolInfo.cs

```diff
@@ -91,35 +72,20 @@ internal SymbolInfo(ISymbol? symbol, ImmutableArray<ISymbol> candidateSymbols, C
             this.CandidateReason = candidateReason;
         }
 
+        internal ImmutableArray<ISymbol> GetAllSymbols()
+            => this.Symbol == null ? CandidateSymbols : ImmutableArray.Create(this.Symbol);
+
         public override bool Equals(object? obj)
-        {
-            return obj is SymbolInfo && Equals((SymbolInfo)obj);
-        }
+            => obj is SymbolInfo info && Equals(info);
 
         public bool Equals(SymbolInfo other)
-        {
-            if (!object.Equals(this.Symbol, other.Symbol) ||
-                _candidateSymbols.IsDefault != other._candidateSymbols.IsDefault ||
-                this.CandidateReason != other.CandidateReason)
-            {
-                return false;
-            }
-
-            return _candidateSymbols.IsDefault || _candidateSymbols.SequenceEqual(other._candidateSymbols);
-        }
+            => this.CandidateReason == other.CandidateReason &&
+               Equals(this.Symbol, other.Symbol) &&
+               CandidateSymbols.SequenceEqual(other.CandidateSymbols);
 
         public override int GetHashCode()
-        {
-            return Hash.Combine(this.Symbol, Hash.Combine(Hash.CombineValues(_candidateSymbols, 4), (int)this.CandidateReason));
-        }
+            => Hash.Combine(this.Symbol, Hash.Combine(Hash.CombineValues(_candidateSymbols, 4), (int)this.CandidateReason));
```

Comment:
`Hash.CombineValues(_candidateSymbols` does the same thing whether or not the immutablearray is default or empty.  I could change this to .CandidateSymbols for consistency though.

Reply:
It feels like too subtle and fragile a distinction as is. It would be better to use `CandidateSymbols`, and let implementation details of `Hash` work themselves out.

Reply:
sounds good.